pkg/render/prometheus_api.go

```diff
@@ -0,0 +1,269 @@
+	// prometheusServiceListenPort defines the port that tigera-prometheus-service
+	// pod listens to and the TargetPort of calico-node-prometheus service.
+	// If not set it defaults to port 9090. In the scenario that the cluster
+	// is hosted on EKS and using Calico as it's CNI, tigera-prometheus-service
+	// will be HostNetwoked. This is configurable in the configMap passed to TigeraPrometheusAPI
+	prometheusServiceListenPort int
+	prometheusServiceImage      string
+}
+
+func (p *tigeraPrometheusAPIComponent) ResolveImages(is *operator.ImageSet) error {
+	reg := p.installation.Registry
+	path := p.installation.ImagePath
+	prefix := p.installation.ImagePrefix
+
+	prometheusServiceImage, err := components.GetReference(components.ComponentTigeraPrometheusService, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+
+	p.prometheusServiceImage = prometheusServiceImage
+
+	return nil
+}
+
+// Objects returns the lists of objects in this component that should be created and/or deleted during
+// rendering.
+func (p *tigeraPrometheusAPIComponent) Objects() (objsToCreate, objsToDelete []client.Object) {
+	// tigera-prometheus-objects
+	namespacedObjects := []client.Object{}
+
+	// place pullsecrets secrets under tigera-prometheus
+	secrets := secret.CopyToNamespace(common.TigeraPrometheusNamespace, p.pullSecrets...)
+	namespacedObjects = append(namespacedObjects, secret.ToRuntimeObjects(secrets...)...)
+
+	namespacedObjects = append(
+		namespacedObjects,
+		p.tigeraPrometheusApiPodSecurityPolicy(),
+		p.tigeraPrometheusServiceDeployment(),
+		p.calicoNodePrometheusService(),
+	)
+
+	objsToCreate = []client.Object{}
+	objsToCreate = append(objsToCreate, namespacedObjects...)
+
+	objsToDelete = []client.Object{}
+
+	return objsToCreate, objsToDelete
+
+}
+
+func (p *tigeraPrometheusAPIComponent) Ready() bool {
+	return true
+}
+
+func (p *tigeraPrometheusAPIComponent) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeLinux
+}
+
+// calicoNodePrometheusService sets up a service for the Prometheus Service/Proxy deployment
+func (p *tigeraPrometheusAPIComponent) calicoNodePrometheusService() *corev1.Service {
+
+	s := &corev1.Service{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Service",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      calicoNodePrometheusServiceName,
+			Namespace: common.TigeraPrometheusNamespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Selector: map[string]string{
+				"k8s-app": tigeraPrometheusAPIName,
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Name:       "web",
+					Port:       common.PrometheusDefaultPort,
+					Protocol:   corev1.ProtocolTCP,
+					TargetPort: intstr.FromInt(p.prometheusServiceListenPort),
+				},
+			},
+		},
+	}
+
+	return s
+}
+
+// tigeraPrometheusServiceDeployment deployment for the Prometheus Service/Proxy pod and image
+func (p *tigeraPrometheusAPIComponent) tigeraPrometheusServiceDeployment() *appsv1.Deployment {
```

Comment:
Please leave Service out of the name here. I'd be fine if this function was reduced to `proxyDeployment`, especially if the component has `tigeraPrometheus` in the name.

Reply:
https://github.com/tigera/operator/pull/1379/commits/0aecaf91feb429b1e0bf6e1e0d25be76e044121e

Reply:
This wasn't shortened any, WDYT about just changing it to deployment?

Reply:
sorry misunderstood last time, I think like the naming of everything related to the componenet to just be its kind, same with PSP below